homeassistant/components/hisense_aehw4a1/__init__.py

```diff
@@ -0,0 +1,77 @@
+"""The Hisense AEH-W4A1 integration."""
+import ipaddress
+
+from pyaehw4a1.aehw4a1 import AehW4a1
+import pyaehw4a1.exceptions
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.components.climate import DOMAIN as CLIMATE_DOMAIN
+from homeassistant.const import CONF_IP_ADDRESS
+import homeassistant.helpers.config_validation as cv
+
+from .const import DOMAIN
+
+
+def coerce_ip(value):
+    """Validate that provided value is a valid IP address."""
+    if not value:
+        raise vol.Invalid("Must define an IP address")
+    try:
+        ipaddress.IPv4Network(value)
+    except ValueError:
+        raise vol.Invalid("Not a valid IP address")
+    return value
+
+
+CONFIG_SCHEMA = vol.Schema(
+    {
+        DOMAIN: {
+            CLIMATE_DOMAIN: vol.Schema(
+                {
+                    vol.Optional(CONF_IP_ADDRESS, default=[]): vol.All(
+                        cv.ensure_list, [vol.All(cv.string, coerce_ip)]
+                    )
+                }
+            )
+        }
+    },
+    extra=vol.ALLOW_EXTRA,
+)
+
+
+async def async_setup(hass, config):
+    """Set up the Hisense AEH-W4A1 integration."""
+    conf = config.get(DOMAIN)
+    hass.data[DOMAIN] = {}
+
+    if conf is not None:
+        devices = conf[CONF_IP_ADDRESS][:]
+        for device in devices:
+            try:
+                await AehW4a1(device).check()
+            except pyaehw4a1.exceptions.ConnectionError:
+                conf[CONF_IP_ADDRESS].remove(device)
+        if conf[CONF_IP_ADDRESS] is not None:
```

Comment:
It will always be a list.
```py
if not conf[CONF_IP_ADDRESS]
```

Reply:
Yes, it checks if the list is empty when you exit the for loop

Reply:
My doubt is on how to prevent automatic discovery if all the IPs have been removed because they have not passed the test

Reply:
The platform will not be loaded of we don't init the config flow, and there is no existing config entry.

Reply:
Please change the code here as my comment above.

Reply:
Ok, I did and pushed. But now it does not work as intended: IP addresses, whether they pass the test or not, are ignored anyway and no integration is created.

Reply:
I changed to `if conf[CONF_IP_ADDRESS]` and now it works

Reply:
Right, that was a type by me.